classes/author/Collector.inc.php

```diff
@@ -1,54 +1,99 @@
 <?php
 /**
- * @file classes/services/QueryBuilders/PKPAuthorQueryBuilder.php
+ * @file classes/author/Collector.inc.php
  *
  * Copyright (c) 2014-2021 Simon Fraser University
  * Copyright (c) 2000-2021 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
- * @class PKPAuthorQueryBuilder
- * @ingroup query_builders
+ * @class author
  *
- * @brief Class for building database queries for authors
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
  */
 
-namespace PKP\services\queryBuilders;
+namespace PKP\author;
 
+use Illuminate\Database\Query\Builder;
 use Illuminate\Support\Facades\DB;
-
+use PKP\core\interfaces\CollectorInterface;
 use PKP\plugins\HookRegistry;
-use PKP\services\queryBuilders\interfaces\EntityQueryBuilderInterface;
 
-class PKPAuthorQueryBuilder implements EntityQueryBuilderInterface
+class Collector implements CollectorInterface
 {
-    /** @var array get authors for one or more contexts */
-    protected $contextIds = [];
+    public const ORDERBY_SEQUENCE = 'a.seq';
+    public const ORDERBY_ID = 'a.author_id';
+
+    /** @var string The default orderBy value for authors collector */
+    public $orderBy = self::ORDERBY_SEQUENCE;
+
+    /** @var DAO */
+    public $dao;
+
+    /** @var array|null */
+    public $contextIds = null;
+
+    /** @var array|null */
+    public $publicationIds = null;
 
     /** @var string get authors with a family name */
     protected $familyName = '';
```

Comment:
These should be initialized to `null`, indicating that the condition should not apply, and the checks for it should be strict around falsy data. See https://github.com/pkp/pkp-docs/issues/812#issuecomment-900639385.

Reply:
(The same applies to other properties here.)

Reply:
Changed